### ISSUE TITLE
odo list showing a duplicate entry of devfile component

### ISSUE THREAD

**Issue by girishramnani** [openshift/odo#4144]
/kind bug

## What versions of software are you using?

**Operating System:**

**Output of `odo version`: odo master

## How did you run odo exactly?
```
girish@MacBook-Pro rtr % odo project create girishn
 ✓  Project 'girishn' is ready for use
 ✓  New project created and now using project: girishn

(base) girish@MacBook-Pro rtr % odo project set girishn   
Already on project : girishn

(base) girish@MacBook-Pro rtr % odo create nodejs --starter --app new-app
...
Starter Project
 ✓  Downloading starter project nodejs-starter from https://github.com/odo-devfiles/nodejs-ex.git [2s]
Please use `odo push` command to create the component with source deployed

(base) girish@MacBook-Pro rtr % odo list
Devfile Components: 
APP         NAME       PROJECT     TYPE       STATE
new-app     nodejs     girishn     nodejs     Not Pushed

(base) girish@MacBook-Pro rtr % odo app list                             
There are no applications deployed in the project 'girishn'

(base) girish@MacBook-Pro rtr % odo push
...
Pushing devfile component nodejs
 ✓  Changes successfully pushed to component
```

## Actual behavior
```
(base) girish@MacBook-Pro rtr % odo list --project girishn --all-apps
Devfile Components: 
APP         NAME       PROJECT     TYPE       STATE
new-app     nodejs     girishn     nodejs     Pushed


Openshift Components: 
APP         NAME       PROJECT     TYPE       SOURCETYPE     STATE
new-app     nodejs     girishn     nodejs                    Pushed
```

as well as --app new-app
```
(base) girish@MacBook-Pro rtr % odo list --project girishn --app new-app
Devfile Components: 
APP         NAME       PROJECT     TYPE       STATE
new-app     nodejs     girishn     nodejs     Pushed


Openshift Components: 
APP         NAME       PROJECT     TYPE       SOURCETYPE     STATE
new-app     nodejs     girishn     nodejs                    Pushed
```

## Expected behavior
```
(base) girish@MacBook-Pro rtr % odo list --project girishn --app new-app
Devfile Components: 
APP         NAME       PROJECT     TYPE       STATE
new-app     nodejs     girishn     nodejs     Pushed
```

## Any logs, error output, etc?
This has been happening after the https://github.com/openshift/odo/pull/4007
And the reason is 
https://github.com/openshift/odo/blob/b1245a533788614a7430b086b663b3a24e3f777f/pkg/component/component.go#L859
`List` function code in component code lists both deployment and deployment Configs. This was done to make the `odo app` implementation streamlined. But due to this `odo list` fetches deployment while it tries to fetch the devfile components and then again fetches deployments and DCs when it lists the components.

Ideally I would prefer that we add a ComponentVersion in the `Component` struct so that I can filter out which components are devfile vs S2I as that is still relavent information that `odo list` cares about.

**Comment by dgolovin** [comment]
@kadel this one and #4136 worth to be fixed and released as 2.0.2 IMHO.

**Comment by dgolovin** [comment]
@girishramnani this is still replicable in 2.0.2

**Comment by kadel** [comment]
> @girishramnani this is still replicable in 2.0.2

Sadly this is not included in 2.0.2.
We had some issues with the release process. It took exceptionally long to push released artifacts :-( 2.0.2 was build from source code before this was merged.
We are currently working on the next release 2.0.3 and that should include this fix.

**Comment by sbouchet** [comment]
verified that issue is fixed in 2.0.3 with redhat-developer/intellij-openshift-connector#251
